test/python/qasm3/test_import.py

```diff
@@ -183,6 +184,47 @@ def dump(self, annotation):  # pylint: disable=redefined-outer-name
         self.assertEqual(qc, expected)
         self.assertTrue(skip_triggered)
 
+    def test_num_qubits(self):
+        """Test num_qubits equal the loaded circuit number of qubits"""
+        program = 'OPENQASM 3.0;\ninclude "stdgates.inc";\nh $0;\ncx $2, $1;\n'
+        out = qasm3.loads(program, num_qubits=5)
+        self.assertEqual(out.num_qubits, 5)
+
+    def test_loads_virtual_qubits(self):
+        """Test num_qubits equal the number of qubits in the loaded circuit having only virtual qubits"""
+        num_qubits = 10
+        qc = QuantumCircuit(num_qubits)
+        for i in range(0, num_qubits, 2):
+            qc.h(i)
+            qc.cx(i, i + 1)
+        qc_ser = qasm3.dumps(qc)
+        qc_unser = qasm3.loads(qc_ser, num_qubits=num_qubits)
+        self.assertTrue(qc_unser.num_qubits <= num_qubits)
```

Comment:
Super minor:
```suggestion
        self.assertLessEqual(qc_unser.num_qubits, num_qubits)
```
but also, should this be a strict equality check, rather than a less-or-equal?

Reply:
made it an equality test with `assertEqual`